src/services/item/plugins/app/appSetting/service.ts

```diff
@@ -76,11 +106,13 @@ export class AppSettingService {
     // delete an app data is allowed to admins
     await this.itemService.get(member, repositories, itemId, PermissionLevel.Admin);
 
+    await this.hooks.runPreHooks('delete', member, repositories, { appSettingId, itemId });
+
```

Comment:
You could move this to the line just below after `const appSetting = await appSettingRepository.get(appSettingId);`. The reasoning is that the get could throw, in which case you would have a state where you have dispatched the pre hooks, but the delete will never even be attempted on the object. I don't think it affects much in the rest of the code but this could help in maintaining consistency.